app/assets/javascripts/angular/directives/pie_chart.js

```diff
@@ -26,20 +26,37 @@ angular.module("Prometheus.directives").directive('pieChart', ["$location", "Wid
           return;
         }
 
+        var expressionIDToLegendID = {};
+        scope.graphSettings.expressions.forEach(function(expr) {
+          expressionIDToLegendID[expr.id] = expr.legendID;
+        });
+
+        legendIDToString = {};
+        scope.graphSettings.legendFormatStrings.forEach(function(ls) {
+          legendIDToString[ls.id] = ls.name;
+        });
+
         pieData.forEach(function(e) {
-          e.value = parseFloat(e.Value || e.value);
+          var m = e.Metric || e.metric;
```

Comment:
I believe we're past the 1 week mark:

https://github.com/prometheus/promdash/commit/6c7d370eae0aa82792fddbf9a687d6a73b1e194f